doc/rst/platforms/infiniband.rst

```diff
@@ -73,56 +92,34 @@ and as an alternative for Slurm, users can write batch submission
 scripts and use ``gasnetrun_ibv`` to launch their jobs.
 
 
----------------------------
-Setting Registration Limits
----------------------------
-
-On most high performance networks, including InfiniBand, memory has
-to be registered with the network in order for Chapel to take
-advantage of fast one-sided communication. On InfiniBand networks
-there may be limits placed on how much memory can be registered so
-GASNet will probe at startup to detect this value. This probing can
-be slow, so GASNet will recommend setting ``GASNET_PHYSMEM_MAX`` to
-avoid probing every time. On nodes with homogeneous amounts of
-memory this message usually looks something like:
-
-.. code-block:: printoutput
-
-      WARNING: Beginning a potentially slow probe of max pinnable memory...
-      WARNING: Probe of max pinnable memory completed in 45s.
-      WARNING:   Probe of max pinnable memory has yielded '335 GB'.
-      WARNING:   If you have the same memory configuration on all nodes, then
-      WARNING:   to avoid this probe in the future either reconfigure using
-      WARNING:      --with-ibv-physmem-max='335 GB'
-      WARNING:   or run with environment variable
-      WARNING:      GASNET_PHYSMEM_MAX='335 GB'.
-
-Where setting ``GASNET_PHYSMEM_MAX='335 GB'`` will quiet the warning
-and skip the startup probe.  On nodes with non-homogeneous amounts
-of memory GASNet may recommend using a fraction of memory instead of
-an absolute value with something like
-``GASNET_PHYSMEM_MAX='0.667'``.
-
-Setting ``GASNET_PHYSMEM_MAX`` to a small value can limit
-communication performance so it is highly recommend to use the value
-GASNet suggests.
-
-
 .. _setting-ibv-spawner:
 
 -------------------
 Selecting a Spawner
 -------------------
 
-Under the covers ``gasnetrun_ibv``-based launchers must figure out
+Under the covers, ``gasnetrun_ibv``-based launchers must figure out
 how to spawn jobs and get them up and running on the compute nodes.
-GASNet's two primary means of doing this on InfiniBand clusters are
-``ssh`` and ``mpi``. GASNet will default to ``mpi`` if MPI support
-is detected at configure time, otherwise it will default to ``ssh``.
-Using ``mpi`` will likely incur a performance penalty because MPI
-will be running concurrently with GASNet. Running with ``ssh`` is
-recommended, but not all systems support ssh'ing to compute nodes so
-it is not always the default.
+GASNet's three ways of doing this on InfiniBand systems are ``pmi``,
+``ssh``, and ``mpi``.  When MPI is detected at configure time, ``mpi``
+will be selected as the default, but we recommend using one of the
+other options if possible.
+
+* ``pmi``: When GASNet's configure step detects a PMI-capable job
+  scheduler like Slurm, ``pmi`` is often the best choice because it
+  typically "just works" and can reduce overhead and configuration
+  steps compared to other options.
```

Comment:
I personally wouldn't put pmi first. The only time I'd try to get pmi working with IB is if I couldn't get ssh working. It can be a pain to locate pmi client headers and ensure that slurm is using a compatible pmi server. And I've seen worse startup performance with pmi compared to ssh. I would put it after ssh but before mpi since mpi can have execution time overhead.

https://pagoda-cxx.slack.com/archives/C01RNPDU0H2/p1617331229135100 has a good summary of some of the issues locating PMI. https://pagoda-cxx.slack.com/archives/C01RNPDU0H2/p1695311923709579 is one example of slow pmi (1024 process startup went from 150s with pmi to 9s with ssh)

Reply:
I'll reorder based on this, thanks @ronawho.  For my own clarity, in your case, was the trick getting GASNet to recognize/find the pmi headers?  Or did the GASNet configure step find them and you still had a hard time getting them to work?

Reply:
[Ah, checking the first slack link you sent, it sounds like it was an issue getting it to work at configure-time]

Reply:
If pmi client headers/libs are installed in standard (default searched) paths, gasnet should automatically pick them up. They're usually not on generic IB systems I've run on, so I needed to build with `export CHPL_GASNET_MORE_CFG_OPTIONS="--with-pmi-home=<PMI_HOME>` so gasnet's configure can pick it up.

Normally I'd try using slurm's pmi (but client libraries aren't always installed by slurm) with something like `--with-pmi-home=<SLURM_HOME>`

(Note that it's been a while since I thought about pmi much and I never really felt all that comfortable anyways. Last time I was wrestling with this I opened https://github.com/Cray/chapel-private/issues/1926 to ask if we could/should bundle pmi to eliminate some of these issues. You still need the system launcher to set up a compatible pmi server, but this would at least eliminate some of the headache of getting client libs/headers.)